zstacklib/zstacklib/utils/test_ipv6_m3_utils.py

```
@@ -0,0 +1,292 @@
+            host = os.environ.get('BM_AGENT_BIND_IP', '::')
+        self.assertEqual(host, 'fd00::1')
+
+    def test_default_double_colon_is_dual_stack(self):
+        """TP-081d: '::' 是 Python/Flask 惯例的双栈监听地址（等价于 0.0.0.0 + ::）"""
+        import socket
+        # '::' 是合法的 IPv6 地址（全零）
+        addr = socket.inet_pton(socket.AF_INET6, '::')
+        self.assertEqual(addr, b'\x00' * 16)
+
+    def test_env_var_removed_restores_default(self):
+        """TP-081e: 删除环境变量后，server['host'] 恢复为 '::'"""
```

Ruff (0.15.11)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)